include/internal/catch_config.cpp

```diff
@@ -9,17 +9,30 @@
 #include "catch_enforce.h"
 #include "catch_stringref.h"
 
+#include <iterator>
+#include <algorithm>
+
 namespace Catch {
 
     Config::Config( ConfigData const& data )
     :   m_data( data ),
         m_stream( openStream() )
     {
+        std::transform(
+            data.testsOrTags.begin(), data.testsOrTags.end(),
+            std::back_inserter( m_testsOrTags ),
+            &trim );
+
+        std::transform(
+            data.sectionsToRun.begin(), data.sectionsToRun.end(),
+            std::back_inserter( m_sectionsToRun ),
+            &trim );
+
         TestSpecParser parser(ITagAliasRegistry::get());
-        if (!data.testsOrTags.empty()) {
+        if (!m_testsOrTags.empty()) {
             m_hasTestFilters = true;
-            for( auto const& testOrTags : data.testsOrTags )
-                parser.parse( testOrTags );
+            for( auto const& testOrTags : m_testsOrTags )
+                parser.parse( trim( testOrTags ) );
```

Comment:
Aren't these already trimmed?

Also now there are helpfully two fields named `testsOrTag` in closely related classes...